File: gadapter/src/main/java/com/zeyad/gadapter/stickyheaders/exposed/StickyHeaderHandler.java
package com.zeyad.gadapter.stickyheaders.exposed;

import android.support.v7.widget.RecyclerView;

import com.zeyad.gadapter.ItemInfo;

import java.util.List;

public interface StickyHeaderHandler {

    /**
     * @return The dataset supplied to the {@link RecyclerView.Adapter}
     */
    List<ItemInfo> getAdapterData();
}

File: gadapter/src/main/java/com/zeyad/gadapter/stickyheaders/exposed/StickyHeaderHandler.kt
package com.zeyad.gadapter.stickyheaders.exposed

import androidx.recyclerview.widget.RecyclerView

import com.zeyad.gadapter.ItemInfo

interface StickyHeaderHandler {

    /**
     * @return The dataset supplied to the [RecyclerView.Adapter]
     */
    val adapterData: List<ItemInfo<*>>
}
